BWEnv/src/controller.cc

```diff
@@ -773,68 +789,88 @@ void Controller::onFrame() {
   // case.
   bool receive_commands = !last_receive_ok;
 
```

Comment:
@jgehring This is new. Experimenting with replacing the full state update serialization with the builder approach. I suspect the latest cause of crashing is that the object API and builder API don't play nicely with each other. So this uses builders all the way down.

Reply:
Confirmed: Builders-all-the-way-down fixed it.